### PR TITLE
add detailed hypervisor and guest OS data

### PR DESCRIPTION
**Pull request by PaulAngus**
## Description

This adds missing hypervisor and guest os data and mappings

## Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

### CODE REVIEW

**Comment by PaulAngus** [comment]
@blueorangutan package

**Comment by PaulAngus** [comment]
@blueorangutan test

**Comment by PaulAngus** [comment]
@blueorangutan package

**Comment by PaulAngus** [comment]
@blueorangutan test

**Comment by rhtyd** [comment]
@blueorangutan package

**Comment by rhtyd** [comment]
@blueorangutan test centos7 vmware-65u2

**Comment by rhtyd** [comment]
@blueorangutan test centos7 xenserver-71

File: engine/schema/src/main/resources/META-INF/db/schema-41200to41300.sql [hunk 1]
```diff
@@ -19,18 +19,154 @@
...
+INSERT IGNORE INTO `cloud`.`hypervisor_capabilities`(uuid, hypervisor_type, hypervisor_version, max_guests_limit, max_data_volumes_limit, max_hosts_per_cluster, storage_motion_supported) values (UUID(), 'XenServer', '7.1.2', 1000, 253, 64, 1);
+
+-- Add VMware 6.7 hypervisor capabilities
+INSERT IGNORE INTO `cloud`.`hypervisor_capabilities`(uuid,hypervisor_type, hypervisor_version, max_guests_limit, security_group_enabled, max_data_volumes_limit, max_hosts_per_cluster, storage_motion_supported, vm_snapshot_enabled) VALUES (UUID(), 'VMware', '6.7', '1024', '0', '59', '64', '1', '1');
+
+-- Update VMware 6.x hypervisor capabilities
+UPDATE `cloud`.`hypervisor_capabilities` SET max_guests_limit='1024', max_data_volumes_limit='59', max_hosts_per_cluster='64' WHERE (hypervisor_type='VMware' AND hypervisor_version='6.0' );
+UPDATE `cloud`.`hypervisor_capabilities` SET max_guests_limit='1024', max_data_volumes_limit='59', max_hosts_per_cluster='64' WHERE (hypervisor_type='VMware' AND hypervisor_version='6.5' );
+
+
```
Comment (rhtyd) [inline engine/schema/src/main/resources/META-INF/db/schema-41200to41300.sql]: Nit - remove extra newline

File: engine/schema/src/main/resources/META-INF/db/schema-41200to41300.sql [hunk 2]
```diff
@@ -19,18 +19,154 @@
...
+INSERT INTO cloud.guest_os (id, uuid, category_id, display_name, created, is_user_defined) VALUES ('298', UUID(), '9', 'FreeBSD 11 (32-bit)', now(), '0');
+INSERT INTO cloud.guest_os (id, uuid, category_id, display_name, created, is_user_defined) VALUES ('299', UUID(), '9', 'FreeBSD 11 (64-bit)', now(), '0');
+INSERT INTO cloud.guest_os (id, uuid, category_id, display_name, created, is_user_defined) VALUES ('300', UUID(), '9', 'FreeBSD 12 (32-bit)', now(), '0');
+INSERT INTO cloud.guest_os (id, uuid, category_id, display_name, created, is_user_defined) VALUES ('301', UUID(), '9', 'FreeBSD 12 (64-bit)', now(), '0');
+INSERT INTO cloud.guest_os (id, uuid, category_id, display_name, created, is_user_defined) VALUES ('302', UUID(), '1', 'CentOS 6.8', now(), '0');
+INSERT INTO cloud.guest_os (id, uuid, category_id, display_name, created, is_user_defined) VALUES ('303', UUID(), '1', 'CentOS 6.9', now(), '0');
+INSERT INTO cloud.guest_os (id, uuid, category_id, display_name, created, is_user_defined) VALUES ('304', UUID(), '1', 'CentOS 6.10', now(), '0');
```
Comment (rhtyd) [inline engine/schema/src/main/resources/META-INF/db/schema-41200to41300.sql]: I'm not sure how to check/test that, we'll need to check the `ID` to be used here. I'm okay if that last ID was 276 in a freshly installed system prior to this change on master/4.11.3.0.

File: engine/schema/src/main/resources/META-INF/db/schema-41200to41300.sql [hunk 3]
```diff
@@ -19,18 +19,154 @@
...
+
+
```
Comment (rhtyd) [inline engine/schema/src/main/resources/META-INF/db/schema-41200to41300.sql]: Minor nit - Remove extra new line.

File: engine/schema/src/main/resources/META-INF/db/schema-41200to41300.sql [hunk 4]
```diff
@@ -19,18 +19,154 @@
...
+INSERT IGNORE INTO `cloud`.`guest_os_hypervisor` (uuid, hypervisor_type, hypervisor_version, guest_os_name, guest_os_id, created, is_user_defined) VALUES (UUID(), 'KVM', 'default', 'Ubuntu 17.04', 277, now(), 0);
+INSERT IGNORE INTO `cloud`.`guest_os_hypervisor` (uuid, hypervisor_type, hypervisor_version, guest_os_name, guest_os_id, created, is_user_defined) VALUES (UUID(), 'KVM', 'default', 'Ubuntu 17.10', 278, now(), 0);
+INSERT IGNORE INTO `cloud`.`guest_os_hypervisor` (uuid, hypervisor_type, hypervisor_version, guest_os_name, guest_os_id, created, is_user_defined) VALUES (UUID(), 'KVM', 'default', 'Ubuntu 18.04 LTS', 279, now(), 0);
+INSERT IGNORE INTO `cloud`.`guest_os_hypervisor` (uuid, hypervisor_type, hypervisor_version, guest_os_name, guest_os_id, created, is_user_defined) VALUES (UUID(), 'KVM', 'default', 'Ubuntu 18.10', 280, now(), 0);
+INSERT IGNORE INTO `cloud`.`guest_os_hypervisor` (uuid, hypervisor_type, hypervisor_version, guest_os_name, guest_os_id, created, is_user_defined) VALUES (UUID(), 'KVM', 'default', 'Ubuntu 19.04', 281, now(), 0);
+
+
```
Comment (rhtyd) [inline engine/schema/src/main/resources/META-INF/db/schema-41200to41300.sql]: Nit - Remove extra newlines

**Comment by rhtyd** [comment]
Xenserver seems happy, kvm tests lgtm

File: engine/schema/src/main/resources/META-INF/db/schema-41200to41300.sql (hunk 4 above)
Comment (PaulAngus) [reply engine/schema/src/main/resources/META-INF/db/schema-41200to41300.sql]: done

File: engine/schema/src/main/resources/META-INF/db/schema-41200to41300.sql (hunk 3 above)
Comment (PaulAngus) [reply engine/schema/src/main/resources/META-INF/db/schema-41200to41300.sql]: done

File: engine/schema/src/main/resources/META-INF/db/schema-41200to41300.sql (hunk 2 above)
Comment (PaulAngus) [reply engine/schema/src/main/resources/META-INF/db/schema-41200to41300.sql]: 276 was last ID in 4.12 release.

**Comment by rhtyd** [comment]
Disk resize failure looks like a possible regression, perhaps needs checking @andrijapanicsb @PaulAngus @borisstoyanov 
@blueorangutan package

**Comment by andrijapanicsb** [comment]
LGTM in regards to OS mappings.

**Comment by rhtyd** [comment]
@blueorangutan package

**Comment by andrijapanicsb** [comment]
> Disk resize failure looks like a possible regression, perhaps needs checking @andrijapanicsb @PaulAngus @borisstoyanov

Seems like an error from  here: https://github.com/apache/cloudstack/blob/1fa4f10a9ea65f156464da1671a4dbf62cc88e06/tools/marvin/marvin/lib/common.py#L1396

Manually repeating the same actions fails with:

**Resizing of root disk is only support for scsi device/bus, the provide VM's disk device bus name is ide0:1** 

Global setting "vmware.root.disk.controller" is set to "ide", instead of "osdefault" which in the end gives "VirtualLsiLogicController" - and then resize works.
@rhtyd /CC @PaulAngus 




vmware.root.disk.controller |  
-- | --




vmware.root.disk.controller |  
-- | --




vmware.root.disk.controller |  
-- | --




vmware.root.disk.controller |  
-- | --

**Comment by rhtyd** [comment]
@blueorangutan package

**Pull request merged**